bitnami/argo-workflows/templates/controller/service.yaml

```diff
@@ -0,0 +1,43 @@
+{{- if or .Values.controller.metrics.enabled .Values.controller.telemetry.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "argo-workflows.controller.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/part-of: argo-workflows
+    app.kubernetes.io/component: controller
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  type: {{ .Values.controller.service.type }}
+  {{- if (or (eq .Values.controller.service.type "LoadBalancer") (eq .Values.controller.service.type "NodePort")) }}
+  externalTrafficPolicy: {{ .Values.controller.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  {{ if eq .Values.controller.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.controller.service.loadBalancerSourceRanges }}
+  {{ end }}
+  {{- if (and (eq .Values.controller.service.type "LoadBalancer") (not (empty .Values.controller.service.loadBalancerIP))) }}
+  loadBalancerIP: {{ .Values.controller.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+  {{- if .Values.controller.metrics.enabled }}
+  - name: {{ .Values.controller.metrics.servicePortName }}
+    port: {{ .Values.controller.metrics.servicePort }}
+    protocol: TCP
+    targetPort: {{ .Values.controller.metrics.port }}
+  {{- end }}
+  {{- if .Values.controller.telemetry.enabled }}
+  - name: {{ .Values.controller.telemetry.servicePortName }}
```

Comment:
Not sure if I see any value in defining the name of the service in the values.yaml file. I imagine this comes from upstream but I don't see why we would want that instead of hardcoding it like the rest.